### PR TITLE
chore(httpClient): add HTTP base with Address retrieve/list resources

### PR DESCRIPTION
**Pull request by dmlittle**
This is the first PR (of many) that includes the new Java wrapper.

**What**
- [x] Add the ability to make GET URL requests with parameters
- [x] Add `RequestOptions` which lets you add a custom API Key, Lob Version or Idempotency Key to each request
- [x] Add the Address model with the ability to retrieve an individual or collection of addresses

### CODE REVIEW

File: src/main/java/com/lob/model/Address.java [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+                ", state='" + state + '\'' +
+                ", country='" + country + '\'' +
+                ", metadata=" + metadata +
+                ", dateCreated=" + dateCreated +
+                ", dateModified=" + dateModified +
+                ", object=" + object +
+                '}';
+    }
+
+
```
Comment (sjl2) [inline src/main/java/com/lob/model/Address.java]: nit: extra line

File: src/main/java/com/lob/model/AddressCollection.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+    @JsonCreator
+    public AddressCollection(
+            @JsonProperty("data") final List<Address> data,
+            @JsonProperty("count") final int count,
+            @JsonProperty("object") final String object) {
+        super(data, count, object);
+    }
+
+
```
Comment (sjl2) [inline src/main/java/com/lob/model/AddressCollection.java]: nit: extra line

File: src/main/java/com/lob/net/APIResource.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    public enum RequestType {
+        NORMAL, MULTIPART
+    }
+
+    public static <T> LobResponse request(APIResource.RequestMethod method,
+                                  String url, Map<String, Object> params, Class<T> clazz,
+                                  RequestOptions options) throws APIException, Exception {
+        return responseGetter.request(method, url, params, clazz, RequestType.NORMAL, options);
```
Comment (sjl2) [inline src/main/java/com/lob/net/APIResource.java]: Is `clazz` because `class` is reserved?

File: src/main/java/com/lob/net/ResponseGetter.java [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+    private static <T> LobResponse _request(APIResource.RequestMethod method,
+                                  String url, Map<String, Object> params, Class<T> clazz,
+                                  APIResource.RequestType type, RequestOptions options) throws AuthenticationException, APIException, IOException {
+        if (options == null) {
+            options = RequestOptions.getDefault();
+        }
+
+        String apiKey = options.getApiKey();
+        if (apiKey == null || apiKey.trim().length() == 0) {
+            throw new AuthenticationException("Missing API Key. (HINT: set your API key using 'Lob.init(<API_KEY>);')", null);
```
Comment (sjl2) [inline src/main/java/com/lob/net/ResponseGetter.java]: nit: Not sure how I feel about about the word `HINT`. I'd expect that in a video game. Maybe something like:

`Missing API Key. Make sure 'Lob.init(<API_KEY>)' is called with a key from your Dashboard.`

File: src/main/java/com/lob/net/APIResource.java (hunk 3 above)
Comment (dmlittle) [inline src/main/java/com/lob/net/APIResource.java]: Correct

**Pull request merged**